src/Microsoft.AspNetCore.Server.Kestrel/Internal/Infrastructure/KestrelThread.cs

```diff
@@ -95,6 +99,10 @@ public void Stop(TimeSpan timeout)
 
             if (_thread.IsAlive)
             {
+                // These operations need to run on the libuv thread so it only makes
+                // sense to attept execution if it's still running
+                DisposeConnections(timeout);
```

Comment:
`timeout` should be `KestrelOptions.ShutdownTimeout` which is "The amount of time after the server begins shutting down before connections will be forcefully closed. Kestrel will wait for the duration of the timeout for any ongoing request processing to complete before terminating the connection. No new connections or requests will be accepted during this time. Defaults to 5 seconds."

`DisposeListeners` should should now probably use a shorter timout (Maybe 1 second?)